### PR TITLE
Fix target installer spec and target methods

### PR DESCRIPTION
**Pull request by dnkoutso**
`target_installer` accepts an instance of `target` but it used methods not present in `target.rb`. The tests themselves used an instance of `pod_target` in order to work which was weird.

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix target installer spec and target methods  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/target.rb [hunk 1]
```diff
@@ -185,12 +216,6 @@ def dummy_source_path
       support_files_dir + "#{label}-dummy.m"
     end
 
-    # @return [String] The version associated with this target
```
Comment (dnkoutso) [inline lib/cocoapods/target.rb]: just moved it further up for organization

File: lib/cocoapods/target/pod_target.rb [hunk 2]
```diff
@@ -191,6 +186,12 @@ def uses_swift?
       end
     end
 
+    # @return [Boolean] Whether the target should build a static framework.
+    #
+    def static_framework?
+      specs.all? { |spec| spec.root.static_framework }
```
Comment (segiddins) [inline lib/cocoapods/target/pod_target.rb]: `root_spec.static_framework`

Comment (dnkoutso) [reply lib/cocoapods/target/pod_target.rb]: fixed

**Pull request merged**